usecase.go

```diff
@@ -82,7 +93,14 @@ func (i UseCase[I, O]) Interactor() usecase.Interactor {
 	return u
 }
 
-func New[I any, O any](input I, output O, interactor UseCaseFunc[I, O], decorationFunc func(IOInteractor *usecase.IOInteractor), m ...Middleware[I, O]) (UseCase[I, O], error) {
+func New[I any, O any](input I,
+	output O,
+	interactor UseCaseFunc[I, O],
+	decorationFunc func(IOInteractor *usecase.IOInteractor),
+	logger log.UseCaseLogger,
```

Comment:
there's a `logging.Logger` interface which you could wrap for your `UseCaseLogger`

Reply:
What package is that referring to? I can't seem to find logging.Logger or `log.Logger`?

Reply:
could have sworn there was one in there